Ignore troublesome paths that cause coverage to fail

asalkeld opened the pull request:
**What this PR does / why we need it**:
`KUBE_COVER=y make check` currently fails, this patch fixes it.

**Which issue this PR fixes** 
fixes #31691

**Special notes for your reviewer**:
None

**Release note**:

``` release-note
NONE
```

This avoids the whole command failing because of errors like the following:

```
# cover k8s.io/kubernetes/pkg/client/restclient
cover: internal error: block 268 overlaps block 270
```

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31949)

<!-- Reviewable:end -->

lavalamp commented:
left a comment on the issue. I don't think skipping this test is a good option? The other skip is because it's plausibly a bug in cover itself. (I don't agree with that skip, either, we should file a bug with the go team)

Commit:
Work around go cover bug

This avoids the whole command failing because of errors like the following:
```
# cover k8s.io/kubernetes/pkg/client/restclient
cover: internal error: block 268 overlaps block 270
```
For vendor code, igone
For local code replace the switch with an if statement

lavalamp commented:
LGTM

asalkeld commented:
@k8s-bot test this

asalkeld commented:
@k8s-bot unit test this
